### PR TITLE
Wildcard SSL certificates

### LINKED ISSUE #18: Wildcard SSL certificates / Better CA ?
* Certificates with wildcards or at least all hostnames ( see http://blog.fealdia.org/2011/12/27/debian-squeeze-self-signed-certificate-for-a-wildcard-commonname/ )
* Use a CA that is recognized by major browsers, or at least write some doc on how to include the authority in them.

### PR DESCRIPTION
**Pull request by krtek4**
PR to fix #18

### CODE REVIEW

**Comment by krtek4** [comment]
@sephii even if I already merged this, could you have a look on the PR to see if I did a blunder ?

Thanks !

File: provisioning/roles/ssl/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ssl_ca_key_file: "{{ ssl_root_dir }}/private/ca.key"
+ssl_ca_csr_file: "{{ ssl_root_dir }}/private/ca.csr"
+ssl_ca_cert_file: "{{ ssl_root_dir }}/ca.crt"
+
+# those you can change
+ssl_conf_file: "{{ ssl_root_dir }}/config/vagrant.cnf"
+ssl_ext_file: "{{ ssl_root_dir }}/config/vagrant.ext"
+ssl_key_file: "{{ ssl_root_dir }}/private/vagrant.key"
+ssl_csr_file: "{{ ssl_root_dir }}/private/vagrant.csr"
+ssl_cert_file: "{{ ssl_root_dir }}/certs/vagrant.pem"
```
Comment (sephii) [inline provisioning/roles/ssl/defaults/main.yml]: Remove trailing white lines

File: provisioning/roles/ssl/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  file: dest={{ ssl_root_dir }}/{{ item }} state=directory
+  with_items: [ certs, db, private, config ]
+  sudo: yes
+
+- name: create database index
+  file: dest={{ ssl_root_dir }}/db/index state=touch
+  sudo: yes
+
+- name: create crlnumber
+  shell: echo 1001 > {{ ssl_root_dir }}/db/crlnumber creates="{{ ssl_root_dir }}/db/crlnumber"
```
Comment (sephii) [inline provisioning/roles/ssl/tasks/main.yml]: Use the copy module (http://docs.ansible.com/ansible/copy_module.html) instead

File: provisioning/roles/ssl/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+- name: create CA key file
+  shell: openssl req -new -config {{ ssl_ca_conf_file }} -out {{ ssl_ca_csr_file }} -keyout {{ ssl_ca_key_file }} creates="{{ ssl_ca_csr_file }}"
+  sudo: yes
+
+- name: create CA cert
+  shell: openssl ca -batch -selfsign -config {{ ssl_ca_conf_file }} -in {{ ssl_ca_csr_file }} -out {{ ssl_ca_cert_file }} -extensions ca_ext creates="{{ ssl_ca_cert_file }}"
+  sudo: yes
+
+
```
Comment (sephii) [inline provisioning/roles/ssl/tasks/main.yml]: Extra whiteline

**Pull request merged**